### ISSUE TITLE
Request: ability to filter out animated emoji

### ISSUE THREAD

**Issue by varjmes** [willmanduffy/mojician#3]
Hi! ✨ 

Firstly, mojician is great! Nice job.

I was wondering if we could create a filter to filter out animated .gifs. Animated .gifs can be a real accessibility hindrance (visual issues, focus issues, vestibular issues) so it'd be nice to only get results for static images as, on the Slack I am on, we don't want animated emoji.

I can help, but I don't know how 💃

**Comment by willmanduffy** [comment]
Totally agree!  I had thought of the accessibility issues myself and wanted to do something around this. I'm currently thinking an option on the account that says "Freeze animated .gifs". Every time a user uploads an emoji I'll process another version using some code like [this](https://makandracards.com/makandra/29813-carrierwave-how-to-remove-gif-animation) (I'm using CarrierWave for image processing).

If a user has that option turned on I'll serve a url with the frozen image instead of the animated one and if they download it to their computer it'll be the frozen version still.

Pros:
- Same number of search results
- Fairly simple to implement

Cons:
- Some freezeframes may be nonsensical because we're just taking the first frame. No context.
- Will require me to regenerate versions of all images in the database which is scary because I'm always worried about messing up the connection to s3 but hey, this is why we have a staging server!

I'm also thinking of adding search filters in general and I could add one to hide or show animated emoji on a search by search basis, but it would default to whatever setting is on the account.

Would love any thoughts / criticisms

**Comment by willmanduffy** [comment]
Hi @Charlotteis! Just released the first version of this that freezes all emoji in search results. I'm going to keep this open because there's 2 more things I want to do before closing this.

1.) An account-wide flag to default to freezing emoji (they can unfreeze anytime they wish!)
2.) Respect freezing the emoji in more places than search results

**Comment by varjmes** [comment]
WillWill you so awesome <3
